enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,278 @@
+
+## Proposal
+
+### Changes on managedClusterSet APIs.
+
+Currently, managedClusterSets use the label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` to select managedClusters. 
+But for some use cases, I want to use other labels to select managedClusters in a managedClusterSets. Meanwhile, managedClusterSets should be exclusive within a purpose, but for different purposes it can overlap.
+So, In this proposal, managedClusterSets could use any labels to select managedClusters. 
+
+So, we add a field `ClusterLabel` into managedClusterSet spec
+
+```go
+ type ManagedClusterSetSpec struct {
+ // ClusterLabel defines the label which the managedClusterSet will use to select managedClusters.
+ // If the field is not set, we will use "cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>" as default value.
+ // +optional
+ ClusterLabel ClusterLabel  `json:"clusterLabel"`
+}
+
+//ClusterLabel defines one cluster label
+type ClusterLabel struct {
+ Key string `json:"key"`
+ Value string `json:"value"`
+}
+```
+
+### RBAC
+Currently, When I want to create a managedClusterSet, I only should have `create` permission to the managedClusterSet. 
+In the managedCluster part, label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` means the cluster is in a managedClusterSet.
+If I want to add the label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` to a managedCluster, I must have `create` permission for subresource `managedclustersets/join`. 
+If I want to add/update other labels, I just need to have `update` permission to the managedCluster.
+
+In this proposal, managedClusterSet could use any labels to select managedClusters. And if I add a label for a managedCluster, it may lead to the managedCluster added to a managedClusterSet. So we need to rethink about the rbac control of managedClusterSet
+
+We will use the following questions to consider the RBAC control.
+
+#### What kinds of roles exist related to managedClusterSets and managedClusters
+In General, there are three kinds of roles related to managedClusterSet.
+1. Cluster admin
+   - Cluster admin could `create`, `get`, `list`, `update`, `delete` all managedClusterSets
+   - Cluster admin could `bind` all managedClusterSets to all namespaces.
+   - Cluster admin could `create`, `get`, `list`, `update`, `delete` all managedClusters
+
+2. Controller/agent which run in managedClusters
+   - The Controller/agent should not `create`, `get`, `list`, `update`, `delete` managedClusterSets
+   - The Controller/agent should not `bind` managedClusterSet.
+   - The Controller/agent should `get`, `update` related managedCluster which is running on.
+
+3. Non Cluster admin users
+   - These users should `get` some managedClusterSets
+   - These users should `bind` some managedClusterSets to their namespaces
+   - These users should `create`, `get`, `update`, `delete` some managedClusters
```

Comment:
This is questionable to me. I think it should callout.

A non-cluster admin user cannot create clusterset.

Also it needs to examplain why 
A non-cluster admin user can create/delete cluster. The update permission should be changed. I do not think a controller or agent or non-cluster admin user can `accept` a cluster.

A controller or agent can update the label of the managed cluster.
A non-cluster user may need to update some cluster's label (not all)

Reply:
done